### DISCUSSION TITLE
Installation error - Knex

### DISCUSSION THREAD

**Discussion by alipasakaratas** [directus/directus#14122] [Archive]
I was doing a clean install, after my successful installation, when I run the npx directus start command, I get the following error.

```
Error: knex: Required configuration option 'client' is missing.
    at new Client (C:\inetpub\vhost\realtygroup.digit.com\RealtyDirectus\node_modules\knex\lib\client.js:58:13)
    at knex (C:\inetpub\vhost\realtygroup.digit.com\RealtyDirectus\node_modules\knex\lib\knex-builder\Knex.js:16:28)
    at getDatabase (C:\inetpub\vhost\realtygroup.digit.com\RealtyDirectus\node_modules\directus\dist\database\index.js:102:32)
    at validateDatabaseConnection (C:\inetpub\vhost\realtygroup.digit.com\RealtyDirectus\node_modules\directus\dist\database\index.js:143:70)
    at createApp (C:\inetpub\vhost\realtygroup.digit.com\RealtyDirectus\node_modules\directus\dist\app.js:91:53)
    at async createServer (C:\inetpub\vhost\realtygroup.digit.com\RealtyDirectus\node_modules\directus\dist\server.js:41:38)
    at async Command.startServer (C:\inetpub\vhost\realtygroup.digit.com\RealtyDirectus\node_modules\directus\dist\server.js:128:20)
    at async Command.parseAsync (C:\inetpub\vhost\realtygroup.digit.com\RealtyDirectus\node_modules\commander\lib\command.js:923:5)
```

### To Reproduce

![1](https://user-images.githubusercontent.com/9849567/175818575-cad5e507-7f7d-4947-b0fb-0d0aada55ba0.png)
![2](https://user-images.githubusercontent.com/9849567/175818576-f2190825-688a-466d-b899-3352dfa95dcf.png)

<details>
<summary>More Info</summary>

### What version of Directus are you using?

9.12.2

### What version of Node.js are you using?

14.16.0

### What database are you using?

Sql Server 17

### What browser are you using?

Chrome, Firefox

### How are you deploying Directus?

Windows server 2016

</details>

**Comment by azrikahar** [comment]
Presumably you were running `npm init directus-project RealtyDirectus` _before_ the first screenshot, what were the exact options you were entering?

The error sounds like something is off with the env configuration. Is there a `.env` file in `C:\inetpub\vhost\realtygroup.rakam.com\RealtyDirectus`? If there is, what is the content of said file?

**Reply by alipasakaratas** [reply]
Our .env file has only the DB credintials, the rest are default 
[file.zip](https://github.com/directus/directus/files/8990564/file.zip)

**Comment by eeerdem** [comment]
Same problem.
I update my directus server form 9.12.0 to 9.12.2, and I tried to solve vulnerabilities by auditing and updating things listed that I didn't know. After that, I tried to start directus and I got same error.

I also tried a clean installation, and I still get same error. It could be becuase of something I updated but I shouldn't.

**Reply by rijkvanzanten** [reply]
> and I tried to solve vulnerabilities by auditing and updating things listed that I didn't know. After that, I tried to start directus and I got same error.

This feels very tricky. What did you update exactly? Randomly updating any dependencies can cause massive issues 👀

**Reply by alipasakaratas** [reply]
Hi,
I'm doing a clean install, still getting the same error. By the way, the new version 9.2.13 has arrived, the same problem continues. I'm getting this error when doing a new installation on 2 different servers.

**Comment by alipasakaratas** [comment]
Hi,

If you try using Mssql Server on your own local computer or server, I think you will definitely get the same error.

**Comment by eeerdem** [comment]
I updated my directus to 9.13.0 and still get the same error.

**Reply by alipasakaratas** [reply]
Hello ,

I found the file that was causing the problem.
Path : ProjectFolder\node_modules\directus\dist\database\index.js

When I replaced the file here with the same file in the old version 9.11.1, the connection was established and the project worked.

**Reply by eeerdem** [reply]
I tested it but doesn't work for me. I have a backup on 9.10 which is working well. But when I try to update it to 9.13, I face same problem :(

**Comment by eeerdem** [comment]
Could it be due to operating system? Is there anybody using 9.12 or 9.13 on Windows Server 2019 without any problem?

**Comment by eeerdem** [comment]
A quick update:
I tested different versions of directus and noticed that until v9.11.1 is woking well. The problem start with v9.12.0 and newer versions.
This should be a bug, or a requirement doesn't match anymore on Windows x64 platform, after v9.12.0

Tested in 3 different PCs with different 64 bit Windows versions; 
Windows 10, Windows 11 and Windows Server 2019

I don't want to open an issue, until one more person test it and share the result here.

**Comment by eeerdem** [comment]
I find the problem and fixed. 
The function getDatabase() in "node_modules/directus/dist/database/index.js" is using get-config-from-env in  "node_modules/directus/dist/utils". 
The function of "transform(key)" in the get-config-from-env is converting letters to camelcase by using "camelcase_1" function which is in the "node_modules/camelcase/index.js".

I log the knexConfig variable and saw that there is a localization problem with Turkish language. DB_CLIENT variable is converted to "clıent" which has a Turkish letter, and should be "client" (as shown in the 1st and 2nd before after photos)

The solution is adding an option of "{locale: false}" in the "transform(key)" in the get-config-from-env file (as shown in the 3rd photo)

I don't know if any other non-English operating systems will also get this problem or not, but adding this extra option is fixed my problem in Turkish operating system. It will be great, if it will be fixed by adding this solution in further versions of directus.

P.S: It took too much time for me to log everthing and find the problem as a beginner programmer :)

![1](https://user-images.githubusercontent.com/108297880/178078174-bf52288b-cd94-4d05-894d-8c1931309883.png)

![2](https://user-images.githubusercontent.com/108297880/178078196-5fbc74e3-eb54-4a3b-b7ef-a80082f4daa2.png)

![3](https://user-images.githubusercontent.com/108297880/178078210-be54086e-14c7-4e6a-97ce-2208e8015ba3.PNG)

**Reply by rijkvanzanten** [reply]
> DB_CLIENT variable is converted to "clıent" which has a Turkish letter, and should be "client" (as shown in the 1st and 2nd before after photos)

What.. Where did that `ı` come from?!

**Reply by eeerdem** [reply]
camelcase function is working by using locale computer language as default, which is Turkish in our case. So, it converts capital "I" to "ı" which is a Turkish letter. But when we add {locale: false} option, it works as English language and converts capital "I" to "i" as expected.

**Reply by rijkvanzanten** [reply]
Damn! Never heard of that before. Lemme create a PR to clean that up :
